build.js

```diff
@@ -0,0 +1,33 @@
+const esbuild = require('esbuild')
```

Comment:
Could you briefly explain why we use `esbuild` instead of `rollup` like we do in other projects?

Reply:
Lot's of reasons here I used esbuild.

1. This project immediately compiles built code from JavaScript to WebAssembly, so there is no need for plugins like banner, and therefore no need to directly use our rollup config.

2. We already use 2 steps of builds, and both have their own compile time, but esbuild comes to rescue because it's far faster than rollup and other bundlers. See this image for how fast they are:
![image](https://github.com/user-attachments/assets/ce090b3d-7365-4b23-8e6d-daea9665712b).
You can get detailed information from their website and repository.

3. Rollup needs external plugins for commonjs transpiler, typescript loader and etc. With esbuild these kind of plugins comes in the box. As you can see in this PR I didn't add any loader or transpiler plugins for esbuild and speed up my development process.

Reply:
1. Do we plan to distribute only wasm file? I also noticed that we generate `build/index.js`
2. Are these tests for random project or our? I mean if it is less than a few seconds and we are not going to grow the project significantly it is not a correct criteria.
3. If an instrument needs plugins to do something it could also be a good thing because in this case esbuild looks like a Swiss knife and it means that we have more hidden deps.

I do not offer you to change the instrument, just noticed weak arguments for exact case.

Reply:
1. There is no need to distribute the build/index.js file because it is of no use; it's just built because js-compute-runtime requires javascript instead of typescript to build. 
2. These benchmark tests are not for our repository but for a medium-weight repository like us.
3. But "esbuild" handles those plugins' maintenance by itself. For example, when we switch to a new version of typescript, we won't need to update plugins and check their compatibility with each other.

Reply:
3. so we use a package with an uncontrolled amount of dependencies that could change at any time? =)

Reply:
3. I think it's not that simple and I think this package has so much users, should strongly depend on semver

Reply:
Instead of using `esbuild` or `rollup`, I suggest a 3rd option: do not use anything at all.

We can rely on running `tsc` to transpile to JS, and `fastly compute build` to generate wasm. [Here](https://github.com/fastly/compute-starter-kit-typescript/blob/main/package.json) is an example Typescript project of Fastly.

The only thing we would lose is the ability to change config store name, right? But on the other hand, we wouldn't need any configuration file, `build.ts` or `rollup.config.js`, no dependency on any build system, we can even remove the entries for webpack from `package.json`. I think this is worth it. And we can use the hardcoded config store name because currently we are not providing the customer to change this name anyway.

Pls let me know what you think @ilfa @Orkuncakilkaya

Reply:
We need to use different Config Store Names for PR E2E Tests and Daily E2E Tests. So we require a build tool or script that will provide us this feature. And the easiest I could find was esbuild. We can switch to rollup for consistency or we can find another way with tsc or something else but IMHO it's really not worth the effort for the case.

Reply:
#1 is okay for me as well

Reply:
#1 is ok for me in this project.

Reply:
Yes, number 1 is good!

Reply:
Thanks for the feedback! @Orkuncakilkaya feel free to resolve the conversation